pilot/pkg/xds/eds.go

```diff
@@ -387,7 +396,12 @@ func (eds *EdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w
 	var edsUpdatedServices map[string]struct{}
 	if !req.Full {
 		edsUpdatedServices = model.ConfigNamesOfKind(req.ConfigsUpdated, gvk.ServiceEntry)
+	} else if len(req.ConfigsUpdated) > 0 && allConfigsWithKind(req.ConfigsUpdated, gvk.ServiceEntry.Kind) {
+		// when full push
+		// if all configs updated are ServiceEntry type, then we only need to send eds for these updated services.
+		edsUpdatedServices = model.ConfigNamesOfKind(req.ConfigsUpdated, gvk.ServiceEntry)
```

Comment:
What about multinetwork Gateways? is that a concern?

Reply:
Should be no. Gateway acts the same as sidecars.